node/sync/src/block_sync.rs

```diff
@@ -319,6 +319,12 @@ impl<N: Network> BlockSync<N> {
             }
             // Update the latest height.
             current_height = self.canon.latest_block_height();
+            // Bump the timestamps of all outstanding requests if we successfully handled a BlockResponse.
+            let mut timestamps = self.request_timestamps.write();
+            let now = Instant::now();
```

Comment:
it'd be best to obtain the time before accessing the lock (as that one involves a system call and would prolong the access)

Reply:
Indeed! Note to self: don't work on PRs on Sunday evening :sweat_smile: